openfisca_core/variables.py

```diff
@@ -301,30 +371,56 @@ def get_formula(self, period = None):
 
         If no period is given and the variable has several formula, return the oldest formula.
 
-        Note: variable.get_formula() returns a simple function, while variable.formula is the subclass of Formula associated to the Variable, a more complex OpenFisca object.
-
         :returns: Formula used to compute the variable
         :rtype: function
         """
 
         if period is None:
-            return self.formula.dated_formulas_class[0]['formula_class'].formula.im_func
+            return self.formulas.peekitem(index = 0)[1] if self.formulas else None
```

Comment:
I had trouble understanding `self.formulas.peekitem(index = 0)[1]`
Would it be interesting to have an intermediate variable such as `oldest_formula` ?

Reply:
Adding a comment